### PR TITLE
[REVIEW] duration - parquet read, write support

### PR DESCRIPTION
**Pull request by karthikeyann**
cuIO parquet reader, writer support for duration types.
addresses part of https://github.com/rapidsai/cudf/issues/5272

supports seconds, milliseconds, microseconds 
supports days, nanoseconds only inside cudf. (parquet 1.0 doesn't support these types)

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

File: cpp/src/io/parquet/writer_impl.cu [hunk 1]
```diff
@@ -166,6 +166,35 @@ class parquet_column_view {
...
+        _physical_type  = Type::INT64;
+        _converted_type = ConvertedType::TIME_MICROS;
+        _stats_dtype    = statistics_dtype::dtype_int64;
+        break;
+      // unsupported outside cudf for parquet 1.0.
+      case cudf::type_id::DURATION_NANOSECONDS:
+        _physical_type  = Type::INT64;
+        _converted_type = ConvertedType::TIME_MICROS;
+        _stats_dtype    = statistics_dtype::dtype_int64;
+        _ts_scale       = -1000;
```
Comment (codereport) [inline cpp/src/io/parquet/writer_impl.cu:196]: Potentially a comment down below on why `-1000` vs `1000` for the `int32_t _ts_scale;` member.

**Comment by revans2** [comment]
This change looks fine to me, but from the documentation it looks like TIME_* was intended to be used for a time without a date, not a time duration.

https://github.com/apache/parquet-format/blob/345282ce307a9c9dcc15b6f3c9106be379ec26ba/LogicalTypes.md#time

Spark uses INTERVAL for all duration values and I don't know what it would do if it encountered a TIME_*, I guess we should test this.
https://github.com/apache/parquet-format/blob/345282ce307a9c9dcc15b6f3c9106be379ec26ba/LogicalTypes.md#interval

TIME_* types according to drill are used for (TIME_MILLIS specifically) "Logical time, not including the date. Annotates int32. Number of milliseconds after midnight." https://drill.apache.org/docs/parquet-format/

The arrow code base indicate this too.
https://github.com/apache/arrow/blob/e2f877cab97d6e0278a91ecd5cd661f3011c7abc/rust/parquet/src/basic.rs#L91-L93

However MATLAB apparently treats them like a duration https://www.mathworks.com/help/matlab/import_export/datatype-mappings-matlab-parquet.html

**Comment by revans2** [comment]
I should add that I am calling this out partly because the spark code cannot use this and we will require support for INTERVAL.  Spark treats INTERVAL types like a struct so perhaps once struct support is in place we can do something to support reading INTERVAL as a struct of 3 duration values and when writing we can detect the same pattern, possibly with a hint.

**Comment by revans2** [comment]
Sorry I am wrong.  Spark does not support storing its CalendarInterval type to parquet.  There was discussion on a JIRA to support it, and the mapping is very clear, but apparently it never was actually done.  So you can ignore all of my comments above.

**Pull request merged**